x/stakeibc/keeper/hooks.go

```diff
@@ -83,10 +84,20 @@ func (k Keeper) BeforeEpochStart(ctx sdk.Context, epochInfo epochstypes.EpochInf
 
 		// Transfers in and out of tokens for hostZones which have community pools
 		k.ProcessAllCommunityPoolTokens(ctx)
+
+		// Do transfers for all reward and swapped tokens defined by the trade routes every stride epoch
+		k.TransferAllRewardTokens(ctx)
 	}
 	if epochInfo.Identifier == epochstypes.MINT_EPOCH {
 		k.AllocateHostZoneReward(ctx)
 	}
+	if epochInfo.Identifier == epochstypes.HOUR_EPOCH {
+		k.UpdateAllSwapPrices(ctx) // update swap spot price frequently
```

Comment:
considering there's a bit of uncertainty w.r.t. the cadence of these two, it might make sense to decouple them and start them both off at the hourly epoch until we look into some more data. If we need to shorten the cadence beyond that, we should probably create a more granular epoch type rather than using an interval to join the two

Reply:
By putting them both in the hourly but possibly allowing them to have different "cadences" like all the examples in the stride epoch above with their own cadences, this should allow for maximum configuration.

Reply:
See comment above, the cadence might actually belong as part of the swap config variables on the tradeRoute

Reply:
I think if we're under the assumption that it's okay if the price is ~1hour stale because of how loose the threshold is, then I think it's simpler to just do these both at the top of the hour

I think should investigate the cadence a bit more with actual data, but until them, I think it makes sense to just keep it as simple as possible

Reply:
from ethan on slack: 
> True totally agree.